client/src/controllers/search.controller.js

```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+app.controller('SearchCtrl', ['$scope', 'SunfishSrvc', function($scope, SunfishSrvc){
+    $scope.search = function() {
+        SunfishSrvc.searchSiafiles($scope.query)
+          .success(function(siafiles){
+              $scope.siafiles = siafiles;
```

Comment:
The thing about CB-hell is when you're in it for so long, you can't see your way out of it. Ask yourself if any of those 3 anonymous functions could be its own named, separated function :P